pyintegration/deephaven2/time.py

```diff
@@ -0,0 +1,775 @@
+
+
+def minute_of_hour(dt, tz: TimeZone) -> int:
+    """ Returns an int value of minutes since the top of the hour for a DateTime in the specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the minutes
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+
+        return _JDateTimeUtils.minuteOfHour(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def month_of_year(dt, tz: TimeZone) -> int:
+    """ Returns an 1-based int value for the month of a DateTime in the specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the month
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.monthOfYear(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def nanos(dt) -> int:
+    """ Returns nanoseconds since Epoch for a DateTime value.
+
+    Args:
+        dt (DateTime): the DateTime for which the nanoseconds offset should be returned
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+
+        return _JDateTimeUtils.nanos(dt)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def nanos_of_day(dt, tz: TimeZone) -> int:
+    """ Returns a long value of nanoseconds since midnight for a DateTime in the specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the nanoseconds since midnight
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.nanosOfDay(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def nanos_of_second(dt, tz: TimeZone) -> int:
+    """ Returns a long value of nanoseconds since the top of the second for a DateTime in the specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the nanoseconds
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+
+        return _JDateTimeUtils.nanosOfSecond(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def nanos_to_millis(ns: int) -> int:
+    """ Converts nanoseconds to milliseconds.
+
+    Args:
+        ns (int): the value of nanoseconds to convert
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.nanosToMillis(ns)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def nanos_to_datetime(ns: int):
+    """ Converts a value of nanoseconds from Epoch to a DateTime.
+
+    Args:
+        ns (long): the long nanoseconds since Epoch value to convert
```

Comment:
```suggestion
        ns (int): nanoseconds since Epoch value to convert
```
Search the docs for "int" and "long" to see what needs to get cleaned up.